util/debug_test.go

```diff
@@ -0,0 +1,179 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"testing"
+
+	"github.com/openconfig/goyang/pkg/yang"
+)
+
+var (
+	debugLibraryOrig    = debugLibrary
+	debugSchemaOrig     = debugSchema
+	maxCharsPerLineOrig = maxCharsPerLine
+	maxValueStrLenOrig  = maxValueStrLen
+)
+
+func saveDebugVals() {
+	debugLibraryOrig = debugLibrary
+	debugSchemaOrig = debugSchema
+	maxCharsPerLineOrig = maxCharsPerLine
+	maxValueStrLenOrig = maxValueStrLen
+}
+
+func restoreDebugVals() {
+	debugLibrary = debugLibraryOrig
+	debugSchema = debugSchemaOrig
+	maxCharsPerLine = maxCharsPerLineOrig
+	maxValueStrLen = maxValueStrLenOrig
+}
+
+func TestDbgPrint(t *testing.T) {
+	saveDebugVals()
+	defer restoreDebugVals()
+
+	debugLibrary = true
+	maxCharsPerLine = 10
+	DbgPrint("test debug <this should be truncated>")
```

Comment:
These tests write to stdout still AFAICS? Did you decide they were worth having?

Reply:
The alternatives would be either dependency injection or changing the behavior through a package global var - but it seems to me these would do more harm than good. At least these tests allow us to visually inspect the output, so we can think of them as example code rather than a real unit test.